### ISSUE TITLE
Meilisearch silently fails to index new documents: no logging, no meaningful response

### ISSUE THREAD

**Issue by doebi** [meilisearch/meilisearch#4438]
**Describe the bug**
When indexing new documents throught the JS api using
```javascript
await client.index('documents').addDocuments([ doc ])
```

meilisearch returns a json telling me about a successfully enqueed task:

```javascript
{ uid: 29, enqueued_at: 2024-02-24 9:56:00.974859795 +00:00:00, started_at: None, finished_at: None, error: None, canceled_by: None, details: Some(DocumentAdditionOrUpdate { received_documents: 1, indexed_documents: None }
```

after a short while the meiliisearch server log prints:

```
meilisearch-1  | [2024-02-24T09:56:00Z INFO  index_scheduler] A batch of tasks was successfully completed.
```

When going to the built-in mini dashboard on port 7700 I see that it created a new index, but the index is empty:
<img width="280" alt="Screenshot 2024-02-24 at 11 02 53" src="https://github.com/meilisearch/meilisearch/assets/1591457/7bcd98ae-8fff-4e31-8f1f-4b07c63de9f2">

**To Reproduce**
Follow my steps as outlined above.

**Expected behavior**

The joke of it is that what this template writes is exactly what i expected from meilisearch: "A clear and concise description of what is failing inside meilisearch while adding my documents."

If the server would tell me whats wrong, I would probably be able to solve it myself. I guess there is something odd with my data, but I have no idea what and trial and error is a tedious and frustrating task, especially in an existing application with real data.

**Screenshots**
If applicable, add screenshots to help explain your problem. (see above)

**Meilisearch version:**
[e.g. v0.20.0]

```
meilisearch-1  | [2024-02-24T09:53:50Z DEBUG index_scheduler] memmap budget: 87960930222080B
meilisearch-1  | [2024-02-24T09:53:50Z DEBUG index_scheduler] index budget: 43958990274560B
meilisearch-1  | [2024-02-24T09:53:50Z DEBUG index_scheduler] Up to 18 indexes of 2199023255552B opened simultaneously.
meilisearch-1  |
meilisearch-1  | 888b     d888          d8b 888 d8b                                            888
meilisearch-1  | 8888b   d8888          Y8P 888 Y8P                                            888
meilisearch-1  | 88888b.d88888              888                                                888
meilisearch-1  | 888Y88888P888  .d88b.  888 888 888 .d8888b   .d88b.   8888b.  888d888 .d8888b 88888b.
meilisearch-1  | 888 Y888P 888 d8P  Y8b 888 888 888 88K      d8P  Y8b     "88b 888P"  d88P"    888 "88b
meilisearch-1  | 888  Y8P  888 88888888 888 888 888 "Y8888b. 88888888 .d888888 888    888      888  888
meilisearch-1  | 888   "   888 Y8b.     888 888 888      X88 Y8b.     888  888 888    Y88b.    888  888
meilisearch-1  | 888       888  "Y8888  888 888 888  88888P'  "Y8888  "Y888888 888     "Y8888P 888  888
meilisearch-1  |
meilisearch-1  | Config file path:	"none"
meilisearch-1  | Database path:		"/data"
meilisearch-1  | Server listening on:	"http://0.0.0.0:7700"
meilisearch-1  | Environment:		"development"
meilisearch-1  | Commit SHA:		"1a083d54fc2840ac59530e5395397211cace35be"
meilisearch-1  | Commit date:		"2024-02-13T14:47:34+00:00"
meilisearch-1  | Package version:	"1.6.2"
meilisearch-1  | Anonymous telemetry:	"Disabled"
meilisearch-1  | Instance UID:		"e194d986-a3fe-4c59-865c-eb646fae9b5d"
meilisearch-1  |
meilisearch-1  | A master key has been set. Requests to Meilisearch won't be authorized unless you provide an authentication key.
meilisearch-1  |
meilisearch-1  | Check out Meilisearch Cloud!	https://www.meilisearch.com/cloud?utm_campaign=oss&utm_source=engine&utm_medium=cli
meilisearch-1  | Documentation:			https://www.meilisearch.com/docs
meilisearch-1  | Source code:			https://github.com/meilisearch/meilisearch
meilisearch-1  | Discord:			https://discord.meilisearch.com
meilisearch-1  |
meilisearch-1  | [2024-02-24T09:53:50Z INFO  actix_server::builder] starting 4 workers
meilisearch-1  | [2024-02-24T09:53:50Z INFO  actix_server::server] Actix runtime found; starting in Actix runtime

```
**Additional context**
Additional information that may be relevant to the issue.
[e.g. architecture, device, OS, browser]

Linux 5.10.0-26-amd64 #1 SMP Debian 5.10.197-1 (2023-09-29) x86_64 GNU/Linux

**Comment by Kerollmops** [comment]
Hey @doebi 👋

Thank you for the issue report. It seems that you understood that Meilisearch sent you a task that has been successfully enqueued however you probably forget to look into [the documentation about tasks processing](https://www.meilisearch.com/docs/learn/async/asynchronous_operations).

Meilisearch processes tasks asynchronously and you can check about their statuses on the `/tasks` route.

I am pretty sure the issue is that the engine is not able to guess the primary key 🤔

Have a good day!

**Comment by doebi** [comment]
@Kerollmops Thanks for the hint with checking `/task`

The information gathered from there is much more informative and should help me fix my setup.

May I suggest to include such a hint in the server logs. Even with the log level at DEBUG all I get is `[2024-02-24T09:55:26Z INFO  index_scheduler] A batch of tasks was successfully completed.`, even tough the task itself failed.

**Comment by curquiza** [comment]
Fixed by #4435, will be release in v1.7.0-rc.2

**Comment by TheDeFiQuant** [comment]
@Kerollmops I am facing the same issue and I agree with @doebi that the application logs should be more verbose in such cases
